<a href="https://colab.research.google.com/github/jailop/notebooks/blob/master/modificaciones_gastos_reservados_2018/notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Modificaciones presupuestarias para gastos reservados de la Presidencia.

Marzo de 2019

En este notebook se analizan las modificaciones presupuestarias relacionadas con la partida de gastos reservados de la Presidencia de la República que fueron realizadas al final del ejercicio fiscal 2018.

La información fuente corresponde a una base de datos proporcionada por el Ministerio de Hacienda en la que están registrados a nivel de objeto específico de gastos las modificaciones hechas al Presupuesto del ejercicio fiscal 2018.

Referencias:

- [Expediente de la solicitud de información de las modificaciones del Presupuesto 2018](https://alac.funde.org/requests/5c4b51a120d8e75a955c8be3)

## 1. Explorando el dataset

En esta sección se hace una revisión general del dataset, para saber su tamaño y los atributos que lo conforman.

In [0]:
# Loading libraries
import pandas as pd

In [0]:
# Loading dataset from Internet and making a local copy
url = 'https://alac.funde.org/docs/5c6346ac3b775031c2865e9c'
data = pd.read_excel(url)
data.to_csv('modificaciones_presupuestarias_2018.csv', index=False)

In [0]:
# Restoring from local copy
data = pd.read_csv('modificaciones_presupuestarias_2018.csv')

In [0]:
# Getting attribute names
data.columns

Index(['EJERCICIO', 'INSTITUCION', 'NOMB_INST', 'AREA_GESTIO',
       'NOMB_AREA_GEST', 'FUENTE_FINANC', 'NOMB_FUEN_FINA', 'CLASIF_PRESUP',
       'NOMB_CLAS_PRES', 'MONTO', 'CLAECO', 'NOMB_CLAS_ECON',
       'TIPO_AFEC_AJUS_PEP', 'AFECTACION', 'ACUERD_EJECUT', 'FECH_ACUE_EJEC',
       'DECRET_LEGISL', 'FECH_DECR_LEGI'],
      dtype='object')

In [0]:
# Getting number of rows and columns
data.shape

(8865, 18)

In [0]:
# Checking head dataset
data.head()

EJERCICIO  INSTITUCION           NOMB_INST  AREA_GESTIO  \
0       2018          100  Organo Legislativo            1   
1       2018          100  Organo Legislativo            1   
2       2018          100  Organo Legislativo            1   
3       2018          100  Organo Legislativo            1   
4       2018          100  Organo Legislativo            1   

              NOMB_AREA_GEST  FUENTE_FINANC NOMB_FUEN_FINA  CLASIF_PRESUP  \
0  Conducción Administrativa              1  Fondo General          51107   
1  Conducción Administrativa              1  Fondo General          51207   
2  Conducción Administrativa              1  Fondo General          51201   
3  Conducción Administrativa              1  Fondo General          51201   
4  Conducción Administrativa              1  Fondo General          51201   

           NOMB_CLAS_PRES      MONTO  CLAECO  NOMB_CLAS_ECON  \
0  Beneficios Adicionales  1800000.0    2111  Remuneraciones   
1  Beneficios Adicionales  1020000.0    2111  Remuneraciones   
2                 Sueldos   282520.0    2111  Remuneraciones   
3                 Sueldos   320525.0    2111  Remuneraciones   
4                 Sueldos   320525.0    2111  Remuneraciones   

   TIPO_AFEC_AJUS_PEP   AFECTACION  ACUERD_EJECUT FECH_ACUE_EJEC  \
0                   1  Disminución         3097.0       27/02/18   
1                   1  Disminución         3097.0       27/02/18   
2                   2      Aumento         3097.0       27/02/18   
3                   2      Aumento         3097.0       27/02/18   
4                   2      Aumento         3097.0       27/02/18   

   DECRET_LEGISL FECH_DECR_LEGI  
0            NaN            NaN  
1            NaN            NaN  
2            NaN            NaN  
3            NaN            NaN  
4            NaN            NaN

El dataset de las modificaciones presupuestarias de 2018 contiene más de 8 mil registros y 18 atributos, entre ellos los referidos a la institución, la clasificación presupuestarias, el monto y las referencias de los documentos de soporte.

In [0]:
# Adjusting sign based on kind of affectation
data['MONTO'] = data['MONTO'] * data['AFECTACION'].apply(lambda x: 1 if x == 'Aumento' else -1)

## 2. Modificaciones de gastos reservados

En esta sección se identifican las modificaciones que corresponden a la partida de gastos reservados.

In [0]:
# Filtering records for code 54315
secret_expenses = data[data['CLASIF_PRESUP'] == 54315]

In [0]:
secret_expenses

EJERCICIO  INSTITUCION                    NOMB_INST  AREA_GESTIO  \
145       2018          500  Presidencia de la República            1   
282       2018          500  Presidencia de la República            1   
283       2018          500  Presidencia de la República            1   
284       2018          500  Presidencia de la República            1   

                NOMB_AREA_GEST  FUENTE_FINANC NOMB_FUEN_FINA  CLASIF_PRESUP  \
145  Conducción Administrativa              1  Fondo General          54315   
282  Conducción Administrativa              1  Fondo General          54315   
283  Conducción Administrativa              1  Fondo General          54315   
284  Conducción Administrativa              1  Fondo General          54315   

        NOMB_CLAS_PRES      MONTO  CLAECO      NOMB_CLAS_ECON  \
145  Gastos Reservados   544434.0    2112  Bienes y Servicios   
282  Gastos Reservados  5764503.0    2112  Bienes y Servicios   
283  Gastos Reservados   800000.0    2112  Bienes y Servicios   
284  Gastos Reservados  3567675.0    2112  Bienes y Servicios   

     TIPO_AFEC_AJUS_PEP AFECTACION  ACUERD_EJECUT FECH_ACUE_EJEC  \
145                   2    Aumento         2009.0       21/12/18   
282                   2    Aumento         1990.0       21/12/18   
283                   2    Aumento         1844.0       05/12/18   
284                   2    Aumento         1929.0       13/12/18   

     DECRET_LEGISL FECH_DECR_LEGI  
145            NaN            NaN  
282            NaN            NaN  
283            NaN            NaN  
284            NaN            NaN

Durante el año 2018 se hiceron 4 modificaciones presupuestarias relacionadas con el objeto específico 54315 - Gastos reservados. Todas corresponden a la Presidencia de la República.

In [0]:
resolutions = secret_expenses.groupby(['ACUERD_EJECUT', 'FECH_ACUE_EJEC']).sum()['MONTO']
resolutions

ACUERD_EJECUT  FECH_ACUE_EJEC
1844.0         05/12/18           800000.0
1929.0         13/12/18          3567675.0
1990.0         21/12/18          5764503.0
2009.0         21/12/18           544434.0
Name: MONTO, dtype: float64

Se observa que las modificaciones relacionadas con la partida de gastos reservados de la Presidencia fueron autorizadas por medio de cuatro acuerdos ejecutivos emitidos durante diciembre de 2018. Al tener estos datos se procedió a solicitar al Ministerio de Hacienda la copia de dichos acuerdos. Los enlaces a los mismos se incluyen a continuación:

- [Acuerdo No. 1844 de Dic. 05, 2018](https://alac.funde.org/docs/5c8038da3b775031c287f990)
- [Acuerdo No. 1929 de Dic 13, 2018](https://alac.funde.org/docs/5c80390f3b775031c287f9a6)
- [Acuerdo No. 1990 de Dic. 21, 2018](https://alac.funde.org/docs/5c8038b93b775031c287f97b)
- [Acuerdo No. 2009 de Dic. 21, 2018](https://alac.funde.org/docs/5c80389d3b775031c287f967)

Referencias:

- [Expediente de la solicitud de información de los acuerdos](https://alac.funde.org/requests/5c756f1c3b775031c2875516)

In [0]:
resolutions.sum() / 1e6

10.676612

En total se hicieron modificaciones por USD 10.7 millones.

## 3. Partidas de origen de las modificaciones

Con base en los números de acuerdos se identifican el resto de partidas que fueron afectadas en las modificaciones de los gastos reservados.

In [0]:
# Getting resolution numbers
resolution_refs = set(resolutions.reset_index()['ACUERD_EJECUT'])
# Filtering for records in resolutions
modif = data[data['ACUERD_EJECUT'].apply(lambda x: x in resolution_refs)]

In [0]:
by_resolution = modif.groupby(['ACUERD_EJECUT', 'CLASIF_PRESUP', 'NOMB_CLAS_PRES']).sum()['MONTO']
by_resolution.to_csv('modificaciones_por_acuerdo.csv', index=False)
by_resolution

ACUERD_EJECUT  CLASIF_PRESUP  NOMB_CLAS_PRES                                   
1844.0         51101          Sueldos                                              -531149.98
               51201          Sueldos                                              -150577.43
               51401          Por Remuneraciones Permanentes                        -41909.22
               51402          Por Remuneraciones Eventuales                          -6364.04
               51501          Por Remuneraciones Permanentes                        -29889.40
               51502          Por Remuneraciones Eventuales                          -9031.17
               51701          Al Personal de Servicios Permanentes                  -28548.60
               51702          Al Personal de Servicios Eventuales                    -2455.69
               51999          Remuneraciones Diversas                                  -74.47
               54315          Gastos Reservados                           

In [0]:
by_code = modif.groupby(['CLASIF_PRESUP', 'NOMB_CLAS_PRES']).sum()['MONTO']
by_code.to_csv('modificaciones_por_especifico.csv', index=False)
by_code

CLASIF_PRESUP  NOMB_CLAS_PRES                                        
51101          Sueldos                                                    -928030.68
51103          Aguinaldos                                                  -35499.60
51201          Sueldos                                                    -730977.94
51203          Aguinaldos                                                  -10221.74
51401          Por Remuneraciones Permanentes                              -85911.86
51402          Por Remuneraciones Eventuales                               -34980.14
51501          Por Remuneraciones Permanentes                              -52118.51
51502          Por Remuneraciones Eventuales                               -44904.83
51701          Al Personal de Servicios Permanentes                        -61724.23
51702          Al Personal de Servicios Eventuales                          -3070.00
51999          Remuneraciones Diversas                                      -183